Content.Server/Security/Systems/DeployableBarrierSystem.cs

```diff
@@ -0,0 +1,43 @@
+using Content.Server.Lock;
+using Content.Server.Storage.Components;
+using Content.Shared.Security;
+using Robust.Server.GameObjects;
+using Robust.Shared.GameObjects;
+using System;
+
+namespace Content.Server.Security.Systems
+{
+    public class DeployableBarrierSystem : EntitySystem
+    {
+        public override void Initialize()
+        {
+            base.Initialize();
+            SubscribeLocalEvent<DeployableBarrierComponent, ComponentStartup>(OnStartup);
+            SubscribeLocalEvent<DeployableBarrierComponent, LockToggledEvent>(OnLockToggled);
+        }
+
+        private void OnStartup(EntityUid uid, DeployableBarrierComponent component, ComponentStartup args)
+        {
+            if (!component.Owner.TryGetComponent(out LockComponent? lockComponent))
+                return;
+
+            ToggleBarrierDeploy(component, lockComponent.Locked);
+        }
+
+        private void OnLockToggled(EntityUid uid, DeployableBarrierComponent component, LockToggledEvent args)
+        {
+            ToggleBarrierDeploy(component, args.Locked);
+        }
+
+        private void ToggleBarrierDeploy(DeployableBarrierComponent component, bool isDeployed)
+        {
+            component.Owner.Transform.Anchored = isDeployed;
+
+            if (!component.Owner.TryGetComponent(out AppearanceComponent? appearanceComponent))
+                return;
+
+            var state = isDeployed ? DeployableBarrierState.Deployed : DeployableBarrierState.Idle;
+            appearanceComponent.SetData(DeployableBarrierVisuals.State, state);
+        }
+    }
```

Comment:
TBH this just seems to double up AnchorableComponent.

Reply:
Could you please give more details? Fyi you can't anchor barrier with wrench.

Reply:
though we don't necessarily have to copy that if we think theres a better way to do it ofc

Reply:
I copied logic from [CEV Eris](https://github.com/discordia-space/CEV-Eris/blob/12a95955fa031ff540b22a536d6e09f881687172/code/game/machinery/deployable.dm#L166). In that codebase wrench is used to repair barriers.

Reply:
that seems fine then for behavior, as for the code structure IMO the ideal would be something like this:
- DeployableBarrierComp/System does not exist, instead replaced with something like LockOnClick and AnchorOnLock (or something like AnchorOnLock which is more inline with your current code, but same effect either way), more modular and can simply
- Lock/Anchorable can set appearance data when lock/anchor is changed themselves, and DeployableBarrierVisualizer can use that instead

Reply:
thistbh, better to genericise behaviours as much as possible as otherwise you drift away from ECS and go more towards actor / OOP stuff.

Reply:
deployable can keep this behavior for now, i will refactor it myself later as part of a more all-encompassing thing